sdk/src/wallet/account/operations/transaction/high_level/send_amount.rs

```diff
@@ -55,17 +97,69 @@ impl AccountHandle {
         options: Option<TransactionOptions>,
     ) -> crate::wallet::Result<PreparedTransactionData> {
         log::debug!("[TRANSACTION] prepare_send_amount");
-        let mut outputs = Vec::new();
+        let rent_structure = self.client.get_rent_structure().await?;
         let token_supply = self.client.get_token_supply().await?;
 
-        for address_with_amount in addresses_with_amount {
-            let (address, bech32_hrp) = Address::try_from_bech32_with_hrp(address_with_amount.address)?;
+        let account_addresses = self.addresses().await?;
+        let default_return_address = account_addresses.first().ok_or(Error::FailedToGetRemainder)?;
+
+        let local_time = self.client.get_time_checked().await?;
+
+        let mut outputs = Vec::new();
+        for AddressWithAmount {
+            address,
+            amount,
+            return_address,
+            expiration,
+        } in addresses_with_amount
+        {
+            let (address, bech32_hrp) = Address::try_from_bech32_with_hrp(address)?;
+            let return_address = return_address
+                .map(|address| Ok::<_, Error>(Address::try_from_bech32_with_hrp(address)?.0))
+                .transpose()?
+                .unwrap_or(default_return_address.address.inner);
             self.client.bech32_hrp_matches(&bech32_hrp).await?;
-            outputs.push(
-                BasicOutputBuilder::new_with_amount(address_with_amount.amount)?
-                    .add_unlock_condition(AddressUnlockCondition::new(address))
-                    .finish_output(token_supply)?,
-            )
+
+            // Get the minimum required amount for an output assuming it does not need a storage deposit.
+            let output = BasicOutputBuilder::new_with_amount(amount)?
+                .add_unlock_condition(AddressUnlockCondition::new(address))
+                .finish_output(token_supply)?;
+            let rent_cost = output.rent_cost(&rent_structure);
```

Comment:
We have this function in the client, which should do the same
```suggestion
            let rent_cost = minimum_storage_deposit_basic_output(&rent_structure, &None, token_supply)?;
```

Reply:
Am I wrong in thinking this calculates the output with a storage deposit as well?

Reply:
It does the same, just less code, but with my other suggestion below this isn't needed at all

Reply:
True, but then I can't use the output below. I would have to create another one.

Reply:
Fair enough